pkg/ccl/crosscluster/physical/standby_read_ts_poller_job.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Licensed as a CockroachDB Enterprise file under the Cockroach Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     https://github.com/cockroachdb/cockroach/blob/master/licenses/CCL.txt
+
+package physical
+
+import (
+	"context"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/jobs"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/security/username"
+	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
+	"github.com/cockroachdb/cockroach/pkg/sql"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/replication"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/errors"
+)
+
+const standbyReadTSPollInterval = 15 * time.Second
+
+type standbyReadTSPollerResumer struct {
+	job *jobs.Job
+}
+
+var _ jobs.Resumer = (*standbyReadTSPollerResumer)(nil)
+
+// Resume implements the jobs.Resumer interface.
+func (r *standbyReadTSPollerResumer) Resume(
+	ctx context.Context, execCtxI interface{},
+) (jobErr error) {
+	execCtx := execCtxI.(sql.JobExecContext)
+	return r.poll(ctx, execCtx.ExecCfg())
+}
+
+// OnFailOrCancel implements the jobs.Resumer interface.
+func (r *standbyReadTSPollerResumer) OnFailOrCancel(
+	ctx context.Context, _ interface{}, jobErr error,
+) error {
+	if jobs.HasErrJobCanceled(jobErr) {
+		err := errors.NewAssertionErrorWithWrappedErrf(
+			jobErr, "standby read ts poller job is not cancelable",
+		)
+		log.Errorf(ctx, "%v", err)
+	}
+	return nil
+}
+
+// CollectProfile implements the jobs.Resumer interface.
+func (r *standbyReadTSPollerResumer) CollectProfile(_ context.Context, _ interface{}) error {
+	return nil
+}
+
+func (r *standbyReadTSPollerResumer) poll(ctx context.Context, execCfg *sql.ExecutorConfig) error {
+	pollInterval := standbyReadTSPollInterval
+
+	if execCfg.StreamingTestingKnobs != nil &&
+		execCfg.StreamingTestingKnobs.StandbyPollInterval.Nanoseconds() > 0 {
+		pollInterval = execCfg.StreamingTestingKnobs.StandbyPollInterval
+	}
+
+	ticker := time.NewTicker(pollInterval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-ticker.C:
+			tenantInfoAccessor := execCfg.UpgradeJobDeps.SystemDeps().TenantInfoAccessor
+			if tenantInfoAccessor == nil {
```

Comment:
i think this should return an assertionFailed error. It's ok to internally fail a non cancellable job if the failure is justified. We can't recover from a nil `TenantInfoAccessor`